### ISSUE TITLE
Type 'AsyncStorageWrapper' is not assignable to type 'PersistentStorage<PersistedData<NormalizedCacheObject>>'

### ISSUE THREAD

**Issue by ball-hayden** [apollographql/apollo-cache-persist#435]
As of `0.10.0`, it isn't possible to use the AsyncStorageWrapper when using TypeScript.

Steps to reproduce:

```
import AsyncStorage from '@react-native-async-storage/async-storage';
import { InMemoryCache } from '@apollo/client';
import { CachePersistor, AsyncStorageWrapper } from 'apollo3-cache-persist';

const cache = new InMemoryCache();

const cachePersistor = new CachePersistor({
  cache,
  storage: new AsyncStorageWrapper(AsyncStorage),
  debounce: 0,
});
```

```
src/modules/graphqlClient/cache.ts(21,3): error TS2322: Type 'AsyncStorageWrapper' is not assignable to type 'PersistentStorage<PersistedData<NormalizedCacheObject>>'.
   Types of property 'setItem' are incompatible.
     Type '(key: string, value: string) => void | Promise<void>' is not assignable to type '(key: string, value: PersistedData<NormalizedCacheObject>) => void | Promise<void> | PersistedData<NormalizedCacheObject> | Promise<...>'.
       Types of parameters 'value' and 'value' are incompatible.
         Type 'PersistedData<NormalizedCacheObject>' is not assignable to type 'string'.
           Type 'null' is not assignable to type 'string'.
```

**Comment by wodCZ** [comment]
Hi @ball-hayden, thanks for the report.

This comes from the same issue already reported in #428, and in more detail in #426.

Basically there was some TypeScript juggling released in 0.10.0, but the actual implementation remained mostly unchanged.
For now you can `@ts-ignore` the line, or stay on 0.9.0. Please subscribe to #426 to get notified when we manage to tackle the issue.

**Comment by ball-hayden** [comment]
Ahh. Apologies for the duplicate then :+1:
